server_test.go

```diff
@@ -59,6 +59,26 @@ func RunLocalUDPServer(laddr string) (*Server, string, error) {
 	return server, l, err
 }
 
+func RunLocalUDPServerWithSecrets(laddr string, secrets map[string]string) (*Server, string, error) {
```

Comment:
I don’t like adding this. Unless it trips the race detector, why not set `TsigSecret` in `TestServerRoundtripTsig`.

Reply:
It tripped the race detector. I couldn't see a way to set TsigSecret under a lock without adding this.

Reply:
I thought it would (I had the same problem in #717). I have a solution that won't blow up the diff though!

Replace:
`func RunLocalUDPServerWithFinChan(laddr string) (*Server, string, chan error, error)`
with
`func RunLocalUDPServerWithFinChan(laddr string, opts ...func(*Server)) (*Server, string, chan error, error)`.

You can then invoke each `opts` before the `ActivateAndServe` goroutine is started, and a closure can be passed in to change fields on `Server`. Because this is a `variadic` function, none of the existing calls have to be updated.

/cc @miekg for his thoughts.

Reply:
That's ultra clever. Implementing it that way.